### PR TITLE
Split Measurable from Node

### PR DESCRIPTION
**Pull request by swankjesse**
The state related to measurements should be its own type.

### CODE REVIEW

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Measurable.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood
+
+public open class Measurable {
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Measurable.kt:18]: This has a naming conflict with Compose. I assume a typealias on that import is enough? Suggestions for other names also welcome!

File: redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/StringWidget.kt [hunk 2]
```diff
@@ -21,29 +21,20 @@ package app.cash.redwood
...
-) {
+) : Measurable() {
   private val words = text.split(" ")
 
   private var left = -1
   private var top = -1
   private var right = -1
   private var bottom = -1
 
-  // TODO: fix the Node API to not conflate layout parameters with measurements.
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonTest/kotlin/app/cash/redwood/StringWidget.kt:32]: Fixing this todo is the big motivation here. StringWidget does all the measurement, Node does all the layout parameters.

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Measurable.kt (hunk 1 above)
Comment (colinrtwhite) [inline redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Measurable.kt:18]: I like this! Later we can add `FlexProperties` so `Node` will just be properties + measurable + layout.

**Review by colinrtwhite — APPROVED** [review]

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt [hunk 3]
```diff
@@ -17,9 +17,9 @@
 
 package app.cash.redwood
```
Comment (JakeWharton) [inline redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt:18]: All of these types need moved into a `flexbox` subpackage at some point.

**Review by JakeWharton — APPROVED** [review]

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt (hunk 3 above)
Comment (JakeWharton) [reply redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt:18]: https://github.com/cashapp/redwood/issues/395

**Pull request merged**